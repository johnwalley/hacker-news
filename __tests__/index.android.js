import 'react-native';
import React from 'react';
import HackerNews from '../App/index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

require('isomorphic-fetch');
require('es6-promise').polyfill();

const response = [{"id":13948658,"title":"3.5 Years, 500k Lines of Go","points":72,"user":"NateDad","time":1490361127,"time_ago":"an hour ago","comments_count":5,"type":"link","url":"https://npf.io/2017/03/3.5yrs-500k-lines-of-go/","domain":"npf.io"},{"id":13948127,"title":"German Space Center has constructed the world’s largest artificial Sun","points":104,"user":"kamaal","time":1490354288,"time_ago":"3 hours ago","comments_count":25,"type":"link","url":"http://newatlas.com/dlr-artificial-sun/48579/","domain":"newatlas.com"},{"id":13948151,"title":"Chasing the First Arcade Easter Egg","points":61,"user":"anjalik","time":1490354601,"time_ago":"3 hours ago","comments_count":5,"type":"link","url":"https://edfries.wordpress.com/2017/03/22/chasing-the-first-arcade-easter-egg/","domain":"edfries.wordpress.com"},{"id":13948900,"title":"Dell’s 32-inch 8K UP3218K Display Now for Sale: Check Your Wallet","points":6,"user":"DiabloD3","time":1490363708,"time_ago":"19 minutes ago","comments_count":0,"type":"link","url":"http://www.anandtech.com/show/11220/dells-32-inch-8k-up3218k-display-now-for-sale","domain":"anandtech.com"},{"id":13947537,"title":"TidalCycles – A language for the Live Coding pattern","points":103,"user":"bojo","time":1490345353,"time_ago":"5 hours ago","comments_count":31,"type":"link","url":"https://tidalcycles.org","domain":"tidalcycles.org"},{"id":13948505,"title":"What’s the Matter with Covert Action?","points":18,"user":"doppp","time":1490359550,"time_ago":"an hour ago","comments_count":3,"type":"link","url":"http://www.filfre.net/2017/03/whats-the-matter-with-covert-action/","domain":"filfre.net"},{"id":13946717,"title":"Tamil Bell","points":285,"user":"Thevet","time":1490331613,"time_ago":"9 hours ago","comments_count":71,"type":"link","url":"https://en.wikipedia.org/wiki/Tamil_bell","domain":"en.wikipedia.org"},{"id":13935486,"title":"What Happens If You Break an Artwork?","points":32,"user":"prismatic","time":1490220734,"time_ago":"2 days ago","comments_count":20,"type":"link","url":"https://www.artsy.net/article/artsy-editorial-break-artwork","domain":"artsy.net"},{"id":13947567,"title":"Ostinato – A Network Traffic Generator and Analyzer","points":54,"user":"rishabhd","time":1490345777,"time_ago":"5 hours ago","comments_count":13,"type":"link","url":"http://ostinato.org/","domain":"ostinato.org"},{"id":13945689,"title":"Angular 4.0.0 Now Available","points":219,"user":"theodorejb","time":1490316933,"time_ago":"13 hours ago","comments_count":228,"type":"link","url":"http://angularjs.blogspot.com/2017/03/angular-400-now-available.html","domain":"angularjs.blogspot.com"},{"id":13945492,"title":"OpenSSL Licensing Update","points":172,"user":"JoshTriplett","time":1490313951,"time_ago":"14 hours ago","comments_count":101,"type":"link","url":"https://www.openssl.org/blog/blog/2017/03/20/license/","domain":"openssl.org"},{"id":13945418,"title":"Sea Ice Extent Sinks to Record Lows at Both Poles","points":170,"user":"matteuan","time":1490313010,"time_ago":"14 hours ago","comments_count":97,"type":"link","url":"https://www.nasa.gov/feature/goddard/2017/sea-ice-extent-sinks-to-record-lows-at-both-poles","domain":"nasa.gov"},{"id":13948172,"title":"iTunes built, and then broke, my meticulous music-listening system","points":47,"user":"anjalik","time":1490354808,"time_ago":"3 hours ago","comments_count":35,"type":"link","url":"http://www.theverge.com/2017/3/23/15031054/how-itunes-built-and-then-broke-my-meticulous-music-listening-system","domain":"theverge.com"},{"id":13938047,"title":"A Russian Poetry Scandal That Ended in a Duel","points":10,"user":"lermontov","time":1490256127,"time_ago":"a day ago","comments_count":0,"type":"link","url":"http://www.atlasobscura.com/articles/russian-literary-hoax-apollo","domain":"atlasobscura.com"},{"id":13941011,"title":"Intent to Deprecate and Remove: Trust in Existing Symantec-Issued Certificates","points":680,"user":"ehPReth","time":1490285130,"time_ago":"a day ago","comments_count":286,"type":"link","url":"https://groups.google.com/a/chromium.org/d/msg/blink-dev/eUAKwjihhBs/rpxMXjZHCQAJ","domain":"groups.google.com"},{"id":13946212,"title":"Drag – Change the way you work inside Gmail","points":56,"user":"nicktimms","time":1490323664,"time_ago":"11 hours ago","comments_count":35,"type":"link","url":"http://www.dragapp.com","domain":"dragapp.com"},{"id":13948228,"title":"Xfce bug: “default desktop screen causes damage to monitor”","points":126,"user":"felipebueno","time":1490355613,"time_ago":"3 hours ago","comments_count":30,"type":"link","url":"https://bugzilla.xfce.org/show_bug.cgi?id=12117","domain":"bugzilla.xfce.org"},{"id":13946258,"title":"Apple says recent Wikileaks CIA docs detail old, fixed iPhone and Mac exploits","points":138,"user":"gerosan","time":1490324377,"time_ago":"11 hours ago","comments_count":47,"type":"link","url":"https://techcrunch.com/2017/03/23/apple-says-recent-wikileaks-cia-docs-detail-old-fixed-iphone-and-mac-exploits/","domain":"techcrunch.com"},{"id":13945591,"title":"Building a Command-Line Application with Crystal","points":89,"user":"willlll","time":1490315226,"time_ago":"14 hours ago","comments_count":27,"type":"link","url":"https://jclem.net/posts/building-a-command-line-application-with-crystal","domain":"jclem.net"},{"id":13944474,"title":"The PhD Octopus (1903)","points":4,"user":"maverick_iceman","time":1490304386,"time_ago":"17 hours ago","comments_count":0,"type":"link","url":"https://www.uky.edu/~eushe2/Pajares/octopus.html","domain":"uky.edu"},{"id":13945842,"title":"Google Team Refines GPU Powered Neural Machine Translation","points":79,"user":"Katydid","time":1490319026,"time_ago":"13 hours ago","comments_count":18,"type":"link","url":"https://www.nextplatform.com/2017/03/20/google-team-refines-gpu-powered-neural-machine-translation/","domain":"nextplatform.com"},{"id":13945473,"title":"Good News on the California Drought","points":182,"user":"hvo","time":1490313715,"time_ago":"14 hours ago","comments_count":109,"type":"link","url":"https://www.nytimes.com/interactive/2017/03/22/us/california-measuring-snowpack.html","domain":"nytimes.com"},{"id":13940412,"title":"Modern JavaScript for Ancient Web Developers","points":553,"user":"rmason","time":1490281945,"time_ago":"a day ago","comments_count":261,"type":"link","url":"https://trackchanges.postlight.com/modern-javascript-for-ancient-web-developers-58e7cae050f9#.2tpky4xfc","domain":"trackchanges.postlight.com"},{"id":13947806,"title":"Code and Fly: Airware (YC W13)","points":null,"user":null,"time":1490349973,"time_ago":"4 hours ago","comments_count":0,"type":"job","url":"item?id=13947806"},{"id":13941923,"title":"Thousands of underground gas bubbles poised to 'explode' in Arctic","points":519,"user":"xg15","time":1490290170,"time_ago":"21 hours ago","comments_count":318,"type":"link","url":"http://siberiantimes.com/science/casestudy/news/n0905-7000-underground-gas-bubbles-poised-to-explode-in-arctic/","domain":"siberiantimes.com"},{"id":13945950,"title":"Twitter explores subscription-based option","points":123,"user":"e15ctr0n","time":1490320583,"time_ago":"12 hours ago","comments_count":117,"type":"link","url":"http://in.reuters.com/article/twitter-subscription-idINKBN16U389","domain":"in.reuters.com"},{"id":13939958,"title":"“Dig once” bill could bring fiber Internet to much of the US","points":438,"user":"walterbell","time":1490278941,"time_ago":"a day ago","comments_count":98,"type":"link","url":"https://arstechnica.com/information-technology/2017/03/nationwide-fiber-proposed-law-could-add-broadband-to-road-projects/","domain":"arstechnica.com"},{"id":13944757,"title":"Going native","points":259,"user":"Garbage","time":1490306491,"time_ago":"16 hours ago","comments_count":186,"type":"link","url":"http://blog.getpostman.com/2017/03/14/going-native/?utm_source=All+contacts&utm_campaign=73d8d3fafd-EMAIL_NEWSLETTER_2017_03_22&utm_medium=email&utm_term=0_bba0cca85f-73d8d3fafd-11457707","domain":"blog.getpostman.com"},{"id":13939645,"title":"Loopy: a tool for thinking in systems","points":463,"user":"bpierre","time":1490276792,"time_ago":"a day ago","comments_count":81,"type":"link","url":"http://ncase.me/loopy/","domain":"ncase.me"},{"id":13938475,"title":"Japanese company develops a solar cell with record-breaking efficiency","points":436,"user":"Jaruzel","time":1490263155,"time_ago":"a day ago","comments_count":102,"type":"link","url":"https://arstechnica.co.uk/science/2017/03/japanese-company-develops-a-solar-cell-with-record-breaking-26-efficiency/","domain":"arstechnica.co.uk"}];

it('renders correctly', () => {
  fetch.mockResponse(JSON.stringify(response));

  const tree = renderer.create(
    <HackerNews />
  );
});
