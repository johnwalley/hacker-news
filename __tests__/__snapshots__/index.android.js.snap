// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column-reverse",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 10,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            dataSource={
              ListViewDataSource {
                "items": 0,
              }
            }
            enableEmptySections={true}
            pageSize={30}
            refreshControl={
              <RefreshControlMock
                onRefresh={[Function]}
                refreshing={true}
              />
            }
            renderFooter={[Function]}
            renderRow={[Function]}
            renderScrollComponent={[Function]}
            renderSeparator={[Function]}
          >
            <RCTRefreshControl />
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderColor": "#8E8E8E",
                      "borderRadius": 5,
                      "borderWidth": 0.5,
                      "opacity": 1,
                      "paddingHorizontal": 20,
                      "paddingVertical": 10,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#8E8E8E",
                      }
                    }
                  >
                    Load More
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
    <View
      index={0}
      layout={
        Object {
          "height": 0,
          "initHeight": 0,
          "initWidth": 0,
          "isMeasured": false,
          "width": 0,
        }
      }
      mode="float"
      navigation={
        Object {
          "dispatch": [Function],
          "goBack": [Function],
          "navigate": [Function],
          "setParams": [Function],
          "state": Object {
            "key": "Init",
            "routeName": "Home",
          },
        }
      }
      navigationState={
        Object {
          "index": 0,
          "routes": Array [
            Object {
              "key": "Init",
              "routeName": "Home",
            },
          ],
        }
      }
      onNavigateBack={[Function]}
      renderLeftComponent={[Function]}
      renderRightComponent={[Function]}
      renderTitleComponent={[Function]}
      router={
        Object {
          "getActionForPathAndParams": [Function],
          "getComponentForRouteName": [Function],
          "getComponentForState": [Function],
          "getPathAndParamsForState": [Function],
          "getScreenConfig": [Function],
          "getStateForAction": [Function],
        }
      }
      style={
        Object {
          "backgroundColor": "#FF6600",
          "elevation": 4,
          "height": 64,
          "paddingTop": 20,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0.5,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 0.5,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "bottom": 0,
                "justifyContent": "center",
                "left": 70,
                "opacity": 1,
                "position": "absolute",
                "right": 70,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, .9)",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "marginHorizontal": 16,
                    "textAlign": "center",
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Top Stories
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
